core/src/test/kotlin/org/eln2/data/SetMapMultiMapTest.kt

```diff
@@ -0,0 +1,286 @@
+package org.eln2.data
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+
+// FIXME: abstract this so that the Mutable version below isn't LITERALLY an unmaintainable copy of this
```

Comment:
The way to do this is to make both interfaces inherit from a super-interface that just defines the common operations, but...

I'm not sure exactly how to do that.

Reply:
You heard my tribulations in real time when I was asking what would happen if you made an "abstract test driver class"; my preferred structure would have been an abstract method to make the instance under test, and the code superficially resembles this right now :)